packages/openapi-typescript/src/index.ts

```diff
@@ -253,7 +256,7 @@ async function openapiTS(schema: string | URL | OpenAPI3 | Readable, options: Op
     output.splice(1, 0, "/** WithRequired type helpers */", "type WithRequired<T, K extends keyof T> = T & { [P in K]-?: T[P] };", "");
   }
 
-  return output.join("\n");
+  return output.join("\n").concat(schemasExportedTypes);
```

Comment:
nit: 

```suggestion
// --root-types
if (options.rootTypes) {
  const schemasExportedTypes = transformComponentsObjectToTypes((allSchemas["."].schema as OpenAPI3).components;
  output.push(schemasExportedTypes);
}

  return output.join("\n");
```

This has the same execution as what you wrote, but it contains all the logic in one place rather than scattering it around in multiple places.

Reply:
I wish I could do this, but `allSchemas` is deleted line 112 of this file 😅 

I imagine you did this on purpose, is there any counter-indication to remove this line ?

Reply:
Sorry; not sure I understand?